src/modules/rlm_smtp/rlm_smtp.c

```diff
@@ -0,0 +1,1107 @@
+	{ FR_CONF_OFFSET("envelope_address", FR_TYPE_STRING, rlm_smtp_t, envelope_address) },
+	{ FR_CONF_OFFSET("recipients", FR_TYPE_TMPL | FR_TYPE_MULTI, rlm_smtp_t, recipient_addrs), .dflt = "&SMTP-Recipients", .quote = T_BARE_WORD},
+	{ FR_CONF_OFFSET("TO", FR_TYPE_TMPL | FR_TYPE_MULTI, rlm_smtp_t, to_addrs), .dflt = "&SMTP-TO", .quote = T_BARE_WORD},
+	{ FR_CONF_OFFSET("CC", FR_TYPE_TMPL | FR_TYPE_MULTI, rlm_smtp_t, cc_addrs), .dflt = "&SMTP-CC", .quote = T_BARE_WORD},
+	{ FR_CONF_OFFSET("BCC", FR_TYPE_TMPL | FR_TYPE_MULTI, rlm_smtp_t, bcc_addrs), .dflt = "&SMTP-BCC", .quote = T_BARE_WORD },
+	{ FR_CONF_OFFSET("timeout", FR_TYPE_TIME_DELTA, rlm_smtp_t, timeout) },
+	{ FR_CONF_OFFSET("set_date", FR_TYPE_BOOL, rlm_smtp_t, set_date), .dflt = "yes" },
+	{ FR_CONF_OFFSET("tls", FR_TYPE_SUBSECTION, rlm_smtp_t, tls), .subcs = (void const *) fr_curl_tls_config },//!<loading the tls values
+	CONF_PARSER_TERMINATOR
+};
+
+/** Iterate over a talloced array of elements
+ *
+@verbatim
+talloc_foreach(vpt_m, tmpl_t *, vpt) {
+	tmpl_debug(vpt);
+}
+@endverbatim
+ *
+ * There seems to be a limitation in for loop initialiser arguments where they all
+ * must be the same type, though we can control the number of layers of pointer
+ * indirection on a per variable basis.
+ *
+ * We declare _p to be a pointer of the specified _type, and initialise it to the
+ * start of the array.  We declare _end to be a pointer of the specified type and
+ * initialise it to point to the end of the array using talloc_array_length().
+ *
+ * _iter is only updated in the condition to avoid de-referencing invalid memory.
+ *
+ * @param[in] _array	to iterate over.  May contain zero elements.
+ * @param[in] _iter	Name of iteration variable.
+ *			Will be declared in the scope of the loop.
+ */
+#define talloc_foreach(_array, _iter) \
+	for (__typeof__(_array[0]) _iter, *_p = (void *)(_array), *_end = (void *)((_array) + talloc_array_length(_array)); \
+	     (_p < _end) && (_iter = *((void **)(_p))); \
+	     _p = (__typeof__(_p))((__typeof__(_array))_p) + 1)
+
+/*
+ * 	Adds every element associated with a dict_attr to a curl_slist
+ */
+static int da_to_slist(fr_mail_ctx *uctx, struct curl_slist **out, const fr_dict_attr_t *dict_attr)
+{
+	REQUEST 			*request = ((fr_mail_ctx *)uctx)->request;
+	VALUE_PAIR			*vp;
+	int 				elems_added = 0;
+
+	/* Iterate over the VP and add the string value to the curl_slist */
+	vp = fr_cursor_iter_by_da_init(&uctx->cursor, &uctx->request->packet->vps, dict_attr);
+	while (vp) {
+		*out = curl_slist_append(*out, vp->vp_strvalue);
+		elems_added++;
+		vp = fr_cursor_next(&uctx->cursor);
+	}
+	/* Check that the elements were found */
+	if (elems_added == 0) {
+		RDEBUG2("There were no %s elements found", dict_attr->name);
+	}
+	return elems_added;
+}
+
+/*
+ *	Adds a comma separated list of emails to a curl slist
+ */
+static int address_to_slist(void *ctx, struct curl_slist **out, const char * address_source, size_t inlen)
```

Comment:
I get what you're trying to do here, and I think it's fine to do format checks, but automatically splitting on commas makes this inconsistent with other areas of the server.

If someone wants to split then they can in theory do `%{explode:, <input>}` to get a list of addresses.  It doesn't quite work yet, but it'll work very soon once I'm done with the code I'm working on.